packages/@ember/-internals/glimmer/tests/integration/components/input-angle-test.js

```diff
@@ -1244,7 +1314,17 @@ moduleFor(
     `[GH#15675] Components test: <Input ${attrs} />`,
     class extends InputRenderingTest {
       renderInput(value = 25) {
-        this.render(`<Input ${attrs.replace('%x', value)} />`);
+        if (attrs.match(/@min|@max/)) {
+          maybeExpectDeprecation(
+            EMBER_MODERNIZED_BUILT_IN_COMPONENTS,
+            () => {
+              this.render(`<Input ${attrs.replace('%x', value)} />`);
+            },
+            /Passing the `@.+` argument to <Input> is deprecated\./
+          );
+        } else {
+          this.render(`<Input ${attrs.replace('%x', value)} />`);
+        }
       }
 
```

Comment:
On second thought, we can do something slightly better for this case:

```ts
class InputAttributesTest extends InputRenderingTest {
  renderInput(value = 25) {
    this.render(`<Input ${attrs.replace('%x', value)} />`);
  }

  ['@test value over default max but below set max is kept']() {
    // ...
  }

  // ...
}

// These are the permutations of the set:
// ['type="range"', 'min="-5" max="50"', value="%x"']

[
  '@type="range" min="-5" max="50" @value="%x"',
  '@type="range" @value="%x" min="-5" max="50"',
  'min="-5" max="50" @type="range" @value="%x"',
  'min="-5" max="50" @value="%x" @type="range"',
  '@value="%x" min="-5" max="50" @type="range"',
  '@value="%x" @type="range" min="-5" max="50"',
].forEach((attrs) => {
  moduleFor(
    `[GH#15675] Components test: <Input ${attrs} />`,
    class extends InputAttributesTest {}
  );
});

[
  '@type="range" @min="-5" @max="50" @value="%x"',
  '@type="range" @value="%x" @min="-5" @max="50"',
  '@min="-5" @max="50" @type="range" @value="%x"',
  '@min="-5" @max="50" @value="%x" @type="range"',
  '@value="%x" @min="-5" @max="50" @type="range"',
  '@value="%x" @type="range" @min="-5" @max="50"',
].forEach((attrs) => {
  moduleFor(
    `[GH#15675] [DEPRECATED] Components test: <Input ${attrs} />`,
    class extends InputAttributesTest {
      renderInput(value = 25) {
        maybeExpectDeprecation(
          EMBER_MODERNIZED_BUILT_IN_COMPONENTS,
          () => super.renderInput(value),
          /Passing the `@(min|max)` argument to <Input> is deprecated\./
        );
      }
    }
  );
});
```

Basically we will define the test class once. Just having the class defined doesn't do anything by itself (it doesn't add the tests to the test suite), it's only when we pass the class to the `moduleFor()` function that actually loops through the methods and find the ones starting with `@test` and pass then to `QUnit.test(...)`.

This allows us to split the existing array into two halves – the deprecated ones and the not deprecated ones. It allows us to give the modules different names (so we can mark it as `[DEPRECATED]`), and avoids the extra duplication in `renderValue` as a bonus.